rasa/nlu/featurizers/count_vectors_featurizer.py

```diff
@@ -220,51 +245,117 @@ def train(
             # create spacy lemma_ for OOV_words
             self.OOV_words = [t.lemma_ for w in self.OOV_words for t in spacy_nlp(w)]
 
-        self.vectorizer = CountVectorizer(
-            token_pattern=self.token_pattern,
-            strip_accents=self.strip_accents,
-            lowercase=self.lowercase,
-            stop_words=self.stop_words,
-            ngram_range=(self.min_ngram, self.max_ngram),
-            max_df=self.max_df,
-            min_df=self.min_df,
-            max_features=self.max_features,
-            analyzer=self.analyzer,
-        )
-
-        lem_exs = [
-            self._get_message_text(example) for example in training_data.intent_examples
-        ]
-
-        self._check_OOV_present(lem_exs)
-
-        try:
-            # noinspection PyPep8Naming
-            X = self.vectorizer.fit_transform(lem_exs).toarray()
-        except ValueError:
-            self.vectorizer = None
-            return
+        if self.use_shared_vocab:
```

Comment:
this should be a separate method `_create_vectorizer`

Reply:
and it should be used in `persist` as well so it should be static function

Reply:
@Ghostvv You mean `load` as well, right?

Reply:
yes